pkg/common/cloudconfig/config_test.go

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package cloudconfig defines azure cloud provider configuration.
+package cloudconfig
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient"
+)
+
+func TestNewCloudConfigFromFile(t *testing.T) {
+	tests := map[string]struct {
+		filePath       string
+		expectErr      bool
+		expectedConfig *CloudConfig
+	}{
+		"file path is empty": {
+			filePath:  "",
+			expectErr: true,
+		},
+		"failed to open file": {
+			filePath:  "./test/not_exist.json",
+			expectErr: true,
+		},
+		"failed to unmarshal file": {
+			filePath:  "./test/azure_config_nojson.txt",
+			expectErr: true,
+		},
+		"failed to validate config": {
+			filePath:  "./test/azure_config_invalid.json",
+			expectErr: true,
+		},
+		"succeeded to load config": {
+			filePath: "./test/azure_config_valid.json",
+			expectedConfig: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud:     "AzurePublicCloud",
+					TenantID:  "00000000-0000-0000-0000-000000000000",
+					UserAgent: "",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "11111111-1111-1111-1111-111111111111",
+					AADClientID:                 "",
+					AADClientSecret:             "",
+				},
+				Location:       "eastus",
+				SubscriptionID: "00000000-0000-0000-0000-000000000000",
+				ResourceGroup:  "test-rg",
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			config, err := NewCloudConfigFromFile(test.filePath)
+			if got := err != nil; got != test.expectErr {
+				t.Errorf("failed to run NewCloudConfigFromFile(%s): got %v, want %v", test.filePath, got, test.expectErr)
```

Comment:
```suggestion
				t.Errorf("Failed to run NewCloudConfigFromFile(%s): got %v, want %v", test.filePath, got, test.expectErr)
```

test errors  should typically be capitalized.
https://google.github.io/styleguide/go/decisions#error-strings